File: springbooto2ehg/read_docx.py
from docx import Document

# 读取docx文件
doc = Document('f:/javapro/bs/springbooto2ehg/需求.docx')

# 遍历所有段落并打印内容
for para in doc.paragraphs:
    text = para.text
    if text:
        print(text)

# 遍历所有表格并打印内容
for table in doc.tables:
    for row in table.rows:
        for cell in row.cells:
            text = cell.text
            if text:
                print(text)

File: springbooto2ehg/ReadDocx.java
import org.apache.poi.xwpf.usermodel.XWPFDocument;
import org.apache.poi.xwpf.usermodel.XWPFParagraph;
import org.apache.poi.xwpf.usermodel.XWPFRun;

import java.io.FileInputStream;
import java.io.IOException;

public class ReadDocx {
    public static void main(String[] args) {
        try {
            FileInputStream fis = new FileInputStream("f:/javapro/bs/springbooto2ehg/需求.docx");
            XWPFDocument document = new XWPFDocument(fis);
            
            for (XWPFParagraph paragraph : document.getParagraphs()) {
                String text = paragraph.getText();
                if (!text.isEmpty()) {
                    System.out.println(text);
                }
            }
            
            document.close();
            fis.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}